File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l = 0
        cols = len(matrix[0])
        rows = len(matrix)
        r = rows * cols - 1
        rowcol = lambda i: (i // cols, i - cols * (i // cols))
        while l <= r:
            center = (l + r) // 2
            row, col = rowcol(center)
            if matrix[row][col] == target:
                return True
            if matrix[row][col] < target:
                l = center + 1
            else:
                r = center - 1

        return False

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size()- 1;

        while (l < r) {
            cout << numbers[l] << " " << numbers[r] << endl;
            if (numbers[l] + numbers[r] == target) {
                return {l + 1, r + 1};
            }
            if (numbers[l] + numbers[r] > target) {
                r--;
            } else {
                l++;
            }
        }

        return {-1, -1};
    }
};
